hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeAdminMonitorImpl.java

```diff
@@ -398,6 +398,12 @@ private boolean checkPipelinesClosedOnNode(TrackedNode dn)
 
   private boolean checkContainersReplicatedOnNode(TrackedNode dn)
       throws NodeNotFoundException {
+    Map<String, List<ContainerID>> containerOnDn = getContainersReplicatedOnNode(dn, true);
+    return (containerOnDn.get("UnderReplicated").size() == 0) && (containerOnDn.get("UnClosed").size() == 0);
+  }
+
+  public Map<String, List<ContainerID>> getContainersReplicatedOnNode(TrackedNode dn, boolean updateMetrics)
```

Comment:
With this change, the client is going to trigger a somewhat expensive operation on SCM. There is potential for multiple clients to issue these calls at the same time, and we don't really have a way to throttle it.

I think it would be better, if `checkContainersReplicatedOnNode` saved the counts and ID lists, perhaps as a map inside TrackedNode. Then `getContainersReplicatedOnNode` can simply retrive the value stored inside tracked node. Note that there will be a period of time where the node is scheduled for decommission, but it has not been checked yet, so we would need to decide what to return in that case.

Reply:
I have followed your suggestion to save the lists of ContainerIDs in TrackedNode as a Map at the end of `checkContainersReplicatedOnNode` and `getContainersReplicatedOnNode` only retrieves this Map.
Currently, for the time period in which the node is scheduled for decommissioning but hasn't been checked yet, an empty map is shown in the place for the container IDs.
Could you please review the PR with these recent changes?